proposals/22866-unified-queue-implementation.md

```diff
@@ -0,0 +1,80 @@
+	* Get summary of software packages installs: https://github.com/fleetdm/fleet/blob/b4a5a1fb49666dd3b10cfd11ccf26190ad9d2902/server/datastore/mysql/software_installers.go#L646
+	* Filter hosts by software install status: https://github.com/fleetdm/fleet/blob/b4a5a1fb49666dd3b10cfd11ccf26190ad9d2902/server/datastore/mysql/software_installers.go#L758
+	* Get host last install data seems to be called for pending installs in some situations: https://github.com/fleetdm/fleet/blob/b4a5a1fb49666dd3b10cfd11ccf26190ad9d2902/server/datastore/mysql/software_installers.go#L783
+	* Batch-set software installers deletes pending installs/uninstalls and marks as "removed": https://github.com/fleetdm/fleet/blob/b4a5a1fb49666dd3b10cfd11ccf26190ad9d2902/server/datastore/mysql/software_installers.go#L858
+	* List host software returns the "latest install attempt", which may be a pending one: https://github.com/fleetdm/fleet/blob/b4a5a1fb49666dd3b10cfd11ccf26190ad9d2902/server/datastore/mysql/software.go#L2218
+	* Setup experience references the `execution_id` of `host_software_installs`, should it bypass the unified queue as this _needs_ to happen during setup, not blocked on any other item that might be in the queue? https://github.com/fleetdm/fleet/blob/b4a5a1fb49666dd3b10cfd11ccf26190ad9d2902/server/datastore/mysql/setup_experience.go#L312
+
+- `host_vpp_software_installs` is used during the "pending" phase:
+	* Filter hosts by software install status : https://github.com/fleetdm/fleet/blob/b4a5a1fb49666dd3b10cfd11ccf26190ad9d2902/server/datastore/mysql/software_installers.go#L705
+	* List host software that is available for install : https://github.com/fleetdm/fleet/blob/b4a5a1fb49666dd3b10cfd11ccf26190ad9d2902/server/datastore/mysql/software.go#L2290
+	* Get summary of VPP software installs: https://github.com/fleetdm/fleet/blob/b4a5a1fb49666dd3b10cfd11ccf26190ad9d2902/server/datastore/mysql/vpp.go#L72
+
+### DB schema changes and migrations
+
+- Entries in the unified queue for a given host will need to be deleted if that host is deleted: https://github.com/fleetdm/fleet/blob/85dd21267700e76ee8fe184ddef2dc54692afe43/server/datastore/mysql/hosts.go#L563
+- Should secondary data be created when creating the unified queue pending record entry? E.g. the `script_contents` row for a script in a software install or script execution request?
+	* I'd say yes, otherwise we could end up taking lots of storage place in this table, defeating the purpose of the `script_contents` deduplication table.
+	* This means that if the script contents are updated, the pending row should be deleted (which I believe is how it works today).
+
+The new table will need to have at least those columns:
+- uuid (unique for each action, will allow cancelation support)
+- host_id (but without FK for performance reasons, as per our DB performance patterns)
+- actor id, email, name and avatar (copied into the table to persist even if user is deleted? a bit like we do for `activities`?)
```

Comment:
During backend sync, we agreed to create a `user_persistent_info` table to handle a deleted user.
https://github.com/fleetdm/fleet/blob/4ee4243f9faccfa190b750c882d2eb68ce83bb71/server/docs/patterns.md?plain=1#L46-L48